pkg/services/accesscontrol/database/externalservices.go

```diff
@@ -10,9 +12,13 @@ import (
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
 )
 
+// extServiceRoleUID generates a 160 bit unique ID using SHA-1 that fits within the 40 characters limit of the role UID.
 func extServiceRoleUID(externalServiceID string) string {
 	uid := fmt.Sprintf("%s%s_permissions", accesscontrol.ExternalServiceRoleUIDPrefix, externalServiceID)
-	return uid
+	hasher := sha1.New()
+	hasher.Write([]byte(uid))
```

Comment:
// Ignore gosec G505 as the hash is not used for security purposes
//nolint:gosec
```suggestion
	hasher := sha1.New()
	_, _ = hasher.Write([]byte(uid))
```

Reply:
Why not return the error in case there is one?